src/iac_scan_runner/scan_runner.py

```diff
@@ -100,49 +128,101 @@ def _init_iac_dir(self, iac_file: UploadFile):
         """
         try:
             iac_filename_local = generate_random_pathname(iac_file.filename)
-            with open(iac_filename_local, 'wb+') as iac_file_local:
+            with open(iac_filename_local, "wb+") as iac_file_local:
                 iac_file_local.write(iac_file.file.read())
                 iac_file_local.close()
             self.iac_dir = unpack_archive_to_dir(iac_filename_local, None)
             remove(iac_filename_local)
         except Exception as e:
-            raise Exception(f'Error when initializing IaC directory: {str(e)}.')
+            raise Exception(f"Error when initializing IaC directory: {str(e)}.")
 
     def _cleanup_iac_dir(self):
         """Remove the created IaC directory"""
         try:
             rmtree(self.iac_dir, True)
         except Exception as e:
-            raise Exception(f'Error when cleaning IaC directory: {str(e)}.')
+            raise Exception(f"Error when cleaning IaC directory: {str(e)}.")
 
-    def _run_checks(self, selected_checks: Optional[List], scan_response_type: ScanResponseType) -> Union[dict, str]:
+    def _run_checks(
+        self, selected_checks: Optional[List], scan_response_type: ScanResponseType
+    ) -> Union[dict, str]:
         """
         Run the specified IaC checks
         :param selected_checks: List of selected checks to be executed on IaC
         :param scan_response_type: Scan response type (JSON or HTML)
         :return: Dict or string with output for running checks
         """
+
+        dt = datetime.now()
+        ts = datetime.timestamp(dt)
+        dir_name = "scan_run_" + str(ts)
+
+        os.mkdir(dir_name)
+
+        compatible_checks = self.checker.get_all_compatible_checks(self.iac_dir)
+        non_compatible_checks = list()
+
         if scan_response_type == ScanResponseType.json:
             scan_output = {}
         else:
             scan_output = ""
         if selected_checks:
             for selected_check in selected_checks:
                 check = self.iac_checks[selected_check]
+                print("NEW CHECK")
+                print(check)
                 if check.enabled:
-                    check_output = check.run(self.iac_dir)
-                    if scan_response_type == ScanResponseType.json:
-                        scan_output[selected_check] = check_output.to_dict()
+                    if selected_check in compatible_checks:
+
+                        check_output = check.run(self.iac_dir)
+
+                        if scan_response_type == ScanResponseType.json:
+                            scan_output[selected_check] = check_output.to_dict()
+                        else:
+                            scan_output += f"### {selected_check} ###\n{check_output.to_string()}\n\n"
+
+                        write_string_to_file(
+                            check.name, dir_name, scan_output[check.name]["output"]
+                        )
```

Comment:
I would not hardcode `"output"`, but rather call an `output` property of `CheckResult`.

```suggestion
                        write_string_to_file(check.name, dir_name, scan_output[selected_check].output)
```

Reply:
Would resolve that later